### PR TITLE
First cut at repl

### PR DESCRIPTION
**Pull request by ejconlon**

### CODE REVIEW

File: lib/Idream/Effects/Idris.hs [hunk 1]
```diff
@@ -77,7 +80,11 @@ idrisGetLibDir = send IdrisGetLibDir
 
 idrisCompile :: Member Idris r
              => ProjectName -> PackageName -> Eff r ()
-idrisCompile projName pkgName = send $ IdrisCompile projName pkgName
+idrisCompile projName pkgName = send $ IdrisCompileLike projName pkgName CompileAction
```
Comment (luc-tielen) [inline lib/Idream/Effects/Idris.hs]: I don't like this approach, splitting up each effect shows more clear intent about what we try to do.
Also compiling and repl are too different to group together under 1 concept.
Better if we use the helper function approach I suggested elsewhere.

File: lib/Idream/Effects/Idris.hs [hunk 2]
```diff
@@ -89,18 +96,24 @@ runIdris f = interpretM g where
         idrisArgs = [ "--libdir" ]
         toDir = filter (/= '\n')
     toDir <$> invokeIdrisWithEnv eh1 eh2 idrisArgs Nothing []
-  g (IdrisCompile projName pkgName) = do
+  g (IdrisCompileLike projName pkgName action) = do
```
Comment (luc-tielen) [inline lib/Idream/Effects/Idris.hs]: Maybe it's better if we create a helper function which the compile and repl case then can call? I was already thinking about refactoring this big function.

File: lib/Idream/Types.hs [hunk 3]
```diff
@@ -94,7 +94,7 @@ data Command = Fetch                        -- ^ Fetches all dependencies as des
              | Compile                      -- ^ Compiles all code (fetch needs to be done first to get dependencies code?)
              | Clean                        -- ^ Cleans up build artifacts, fetched code?
              | Run [Argument]               -- ^ Runs the executable defined in idr-project.json
-             | Repl                         -- ^ Opens up the repl
+             | Repl ProjectName PackageName  -- ^ Opens up the repl
```
Comment (luc-tielen) [inline lib/Idream/Types.hs:97]: Why did you make it so we have to provide project name and package name?
I haven't messed around with REPL command too much.. but I would imagine mainly loading the main packages we are working on? (the current 'project' we are working on)?

**Review by luc-tielen — CHANGES_REQUESTED** [review]
For a first start this looks really good!
I can imagine that the freer approach was mind bending at first.. 😅

I would split up the effects for repl and compile though, and reuse common functionality with a helper function.

**Review by luc-tielen — APPROVED** [review]

**Comment by luc-tielen** [comment]
I ended up doing the changes myself, was too enthousiastic and couldn't resist tinkering with it myself.. merging it in now.

File: lib/Idream/Types.hs (hunk 3 above)
Comment (ejconlon) [reply lib/Idream/Types.hs:97]: Not sure how to determine what's the "main" package.  (Also I marked the PR WIP for Work-In-Progress)

**Pull request merged**